lib/evaluator/handlers.js

```diff
@@ -81,16 +81,47 @@ exports.FilterExpression = function(ast) {
  * @private
  */
 exports.Identifier = function(ast) {
+  function contextHasProp(context, prop) {
+    // If this is primitive object (eg "string") we can't check
+    // this, which is why we have to be careful not to look in things
+    // that aren't objects.
+    if (typeof context == "object" && !(prop in context)) {
```

Comment:
Object.prototype.hasOwnProperty?

Reply:
Good call.  The reason I didn't do this before is because the m-c code uses proxies, and I didn't realize that it was possible to trap Object.prototype.hasOwnProperty. I just did some more research and see that it is, so I'll go ahead and make this change.